test/chtest-make-auto

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+fatal () {
+    printf "error: %s\n" "$1" >&2
+    exit 1
+}
+
+tag_from_path () {
+    parent_dir=$(dirname "$path_")
+    basicname=${basename_%%.*}
+    extension=${basename_##*.}
+
+    if [[ $extension == "$basicname" ]]; then
+        extension=''
+    else
+        extension=${extension/\./} # remove dot
+    fi
+
+    case $basicname in
+    Build|Dockerfile|Docker_Pull)
+        if [[ $parent_dir == '' ]]    || \
+           [[ $parent_dir == '.' ]]   || \
+           [[ $parent_dir == test ]]; then
+            if [[ -n $extension ]]; then
+                echo "$extension"
+            fi
+        else
+            if [[ -z $extension ]]; then
+                echo $(basename "$parent_dir")
+            else
+                echo $(basename "${parent_dir}-${extension}")
+            fi
+        fi
+    ;;
+    *)
+        fatal "chtest-make-auto: invalid basic name '$basicname'"
+    esac
+}
+
+template_print () {
+    template="./make-auto.d/$1"
+
+    cat "$template" | sed "s@%(addr)s@$addr@g" \
+                    | sed "s@%(basename)s@$basename_@g" \
+                    | sed "s@%(dirname)s@$dirname_@g" \
+                    | sed "s@%(path)s@$path_@g" \
+                    | sed "s@%(scope)s@$scope@g" \
+                    | sed "s@%(tag)s@$tag@g"
```

Comment:
I think you can reduce this to a single `sed` or `awk` invocation, and remove `cat`.